ci-operator/step-registry/ipi/conf/aws/min-perms-user/ipi-conf-aws-min-perms-user-commands.sh

```diff
@@ -0,0 +1,131 @@
+trap 'rm -rf /tmp/aws_cred_output /tmp/pull-secret /tmp/min_perms/' EXIT TERM INT
+
+function run_command() {
+    local cmd="$1"
+    echo "Running Command: ${cmd}"
+    eval "${cmd}"
+}
+
+function installer_generate_policies()
+{
+	local dir=/tmp/min_perms/
+
+	mkdir -p ${dir}
+
+	# Make a copy of the install-config.yaml since the installer will consume it.
+	cp "${SHARED_DIR}/install-config.yaml" ${dir}/
+
+	cmd="openshift-install create permissions-policy --dir ${dir}"
+	run_command "${cmd}" || return 1
+
+	# Save policies to shared dir so later steps have access to it.
+	mv ${dir}/*-creds.json ${SHARED_DIR}/
+
+	return 0
+}
+
+function aws_create_policy()
+{
+    local aws_region=$1
+    local policy_name=$2
+    local policy_doc=$3
+    local output_json="$4"
+
+    cmd="aws --region $aws_region iam create-policy --policy-name ${policy_name} --policy-document '${policy_doc}' > '${output_json}'"
+    run_command "${cmd}" || return 1
+    return 0
+}
+
+function aws_create_user()
+{
+    local aws_region=$1
+    local user_name=$2
+    local policy_arn=$3
+    local user_output=$4
+    local access_key_output=$5
+
+    # create user
+    cmd="aws --region ${aws_region} iam create-user --user-name ${user_name} > '${user_output}'"
+    run_command "${cmd}" || return 1
+
+    # attach policy
+    cmd="aws --region ${aws_region} iam attach-user-policy --user-name ${user_name} --policy-arn '${policy_arn}'"
+    run_command "${cmd}" || return 1
+
+    # create access key
+    cmd="aws --region ${aws_region} iam create-access-key --user-name ${user_name} > '${access_key_output}'"
+    run_command "${cmd}" || return 1
+
+    return 0
+}
+
+if [[ "${AWS_INSTALL_USE_MINIMAL_PERMISSIONS}" != "yes" ]]; then
+	echo "Custom AWS user with minimal permissions is disabled. Using AWS user from cluster profile."
+	exit 0
+fi
+
+RELEASE_IMAGE_INSTALL="${RELEASE_IMAGE_INITIAL:-}"
+if [[ -z "${RELEASE_IMAGE_INSTALL}" ]]; then
+  # If there is no initial release, we will be installing latest.
+  RELEASE_IMAGE_INSTALL="${RELEASE_IMAGE_LATEST:-}"
+fi
+cp ${CLUSTER_PROFILE_DIR}/pull-secret /tmp/pull-secret
+oc registry login --to /tmp/pull-secret
+ocp_version=$(oc adm release info --registry-config /tmp/pull-secret ${RELEASE_IMAGE_INSTALL} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
+ocp_major_version=$( echo "${ocp_version}" | awk --field-separator=. '{print $1}' )
+ocp_minor_version=$( echo "${ocp_version}" | awk --field-separator=. '{print $2}' )
+rm /tmp/pull-secret
+
+if (( ocp_major_version < 4 || ( ocp_major_version == 4 && ocp_minor_version < 18 ))); then
+  echo "Custom AWS user with minimal permissions is not supported in target latest version ${ocp_version}; bypassing configuration"
+  exit 0
+fi
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+REGION="${LEASED_RESOURCE}"
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+installer_generate_policies
```

Comment:
QE developed a step to provision a minimal permission user to run install, like `aws-provision-iam-user-minimal-permission`. But we were using a static permission policy list, now sounds 4.18 installer would have a new command to create the permission policy list by `openshift-install create permissions-policy`. 

Shall we merge the new command to `aws-provision-iam-user-minimal-permission`, so that we can keep the same step to cover the minimal permission user installation on all versions? cc @yunjiang29

Reply:
@jianlinliu I was thinking about doing that as a follow-up, once we have the `create permissions-policy` working. My idea was to merge these changes first so I can test the installer PR introducing the new command at https://github.com/openshift/installer/pull/8704

Reply:
Sounds a plan for me.

Reply:
sounds good to me